packages/blade/src/components/Tour/_decisions/decisions.md

```diff
@@ -0,0 +1,766 @@
+    - [Conclusion](#conclusion)
+- [MoMs](#moms)
+- [References](#references)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=63871%3A13263&mode=dev) to all variants of the Tour component
+
+## Features
+
+- [x] **Tour:** A guided tour with multiple steps
+- [x] **Masking:** Mask the rest of the page except the highlighted element.
+
+## Basic Usage Structure
+
+With Tour component, you can specify an array of steps with a `name` prop for each step, and then use the `TourStep` component to wrap the element that needs to be highlighted with the same `name` prop.
+
+The Tour component internally handles everything for your
+
+- Masking of the highlighted component
+- Internal state management of `activeStep`
+- Navigation between steps
+
+> Check the full example on [Usage](#usage) section below
+
+```jsx
+const steps = [
+  {
+    name: 'step-1',
+    title: 'Step 1',
+    content: () => <Text>Some content for step 1</Text>,
+    footer: () => <Text>Footer for step 1</Text>,
+  },
+  // ...more steps
+];
+
+<Tour steps={steps}>
+  // ... more jsx
+  <TourStep name="step-1">
+    <Box>I'll be highlighted</Box>
+  </TourStep>
+</Tour>;
+```
+
+## API
+
+### `Tour` API
+
+The `Tour` component is used to render a guided tour with multiple steps.
+Each tour `step` will have a subset of props from the `Popover` component, with extra props for the tour related logic.
+
+> Check the Popover API Decisions [here](https://github.com/razorpay/blade/blob/master/packages/blade/src/components/Popover/_decisions/decisions.md)
+
+```jsx
+// Step will have similar props as the Popover component, With extra Tour related props.
+type Step = {
+  /**
+   * Unique identifier for the tour step
+   */
+  name: string;
+  /**
+   * Content of the Popover
+   */
+  content: ({ gotToNext, goToPrev, stopTour, activeStep, totalSteps }) => React.ReactNode;
+  /**
+   * Footer content
+   */
+  footer?: ({ gotToNext, goToPrev, stopTour, activeStep, totalSteps }) => React.ReactNode;
+  /**
+   * Popover title
+   */
+  title?: string;
+  /**
+   * Leading content placed before the title
+   *
+   * Can be any blade icon or asset.
+   */
+  titleLeading?: React.ReactNode;
+  /**
+   * Placement of Popover
+   * @default "top"
+   */
+  placement?: UseFloatingOptions['placement'];
+}
+
+type TourProps = {
+  /**
+   * Array of steps to be rendered
+   *
+   * The order of the steps will be the order in which they are rendered depending on the `activeStep` prop
+   */
+  steps: Step[];
+  /**
+   * Whether the tour is visible or not
+   */
+  isOpen: boolean;
+  /**
+   * Callback when the tour is opened or closed
+   */
+  onOpenChange: ({ isOpen: boolean }) => void;
+  /**
+   * Callback which fires when the tour has reached the last step.
+   */
+  onFinish: () => void;
```

Comment:
Seems bit redundant with `onStepChange` in the picture

Reply:
Actually, the JSDOC is wrong. 

This was bit of a gray area in terms of differences between onStepChange vs onFinish when i started writing the API. 

But after I've done the implementation it's more clear now, 
The usecase is creating interruptible tours, this is an existing usecase in Dashboard:  

There can be possibility that you need to interrupt the tour or skip the tour in the middle, so when when you call `stopTour` function from the `steps` render prop only then this onFinish will get called. 

```
const steps = [
  {
    name: 'step-1',
    content: ({ stopTour }) => <Button onClick={stopTour}>Skip Tour</Button>
  }
]

<Tour onFinish={() => "Called when stopTour is called"} />
```

Check this example of [interruptible tour](https://github.com/razorpay/blade/pull/1802/files#diff-2ad1c41c9aaa02282767d4d760cd3ff77f7d54c09b0b905a2b866be807b04a33R908-R994)